public/app/core/services/backend_srv.ts

```diff
@@ -142,6 +142,53 @@ export class BackendSrv implements BackendService {
     });
   }
 
+  lines(options: BackendSrvRequest): Observable<string> {
+    return new Observable((observer) => {
+      let done = false;
+      const sub = this.fetch<ReadableStream<Uint8Array>>({
+        ...options,
+        responseType: 'body',
+        showErrorAlert: false,
+        showSuccessAlert: false,
+        hideFromInspector: true,
+      }).subscribe((result) => {
+        if (!result.ok) {
+          observer.error(result);
+          return;
+        }
+
+        const reader = result.data.getReader();
+        async function process() {
+          while (!done) {
+            const chunk = await reader.read();
+            if (chunk.value) {
+              const buff = new TextDecoder().decode(chunk.value);
+              buff.split('\n').forEach((v) => {
```

Comment:
@Clarity-89 the [official k8s js client](https://github.com/kubernetes-client/javascript/blob/main/src/watch.ts#L61) uses:
https://nodejs.org/api/readline.html#readlinepromisescreateinterfaceoptions

I'm not sure if we need the fancy json parsing we have in:
https://github.com/grafana/grafana/blob/grafana-git-ui-sync/public/app/features/apiserver/client.ts#L76

Reply:
Yeah, let's uses whatever is simpler.

Reply:
After thinking on this more...  maybe `backendSrv` should return the raw value -- not string, so the details of each chunk are handled by the consumer

Reply:
Could do that and separately add a function/method to process those values.

Reply:
I pushed an update and renamed the function to `chunked()`